### PR TITLE
New integration BeyondTrust Password Safe

### PR DESCRIPTION
**Pull request by Shellyber**
## Status
In Progress

## Related Issues
fixes: https://github.com/demisto/etc/issues/16536

### CODE REVIEW

**Comment by Shellyber** [comment]
Hi @kirbles19 please take a look at the yml file and the MD file.

**Comment by kirbles19** [comment]
@Shellyber Finished updating .yml, .md, and .py files.

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 1]
```diff
@@ -0,0 +1,316 @@
...
+import requests
+from requests import Request, Session
+
+
+# Disable insecure warnings
+requests.packages.urllib3.disable_warnings()
+
+''' GLOBALS/PARAMS '''
+
+USERNAME = demisto.params().get('credentials').get('identifier')
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
USERNAME = demisto.params().get('credentials', {}).get('identifier')
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 2]
```diff
@@ -0,0 +1,316 @@
...
+PASSWORD = demisto.params().get('credentials').get('password')
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
PASSWORD = demisto.params().get('credentials', {}).get('password')
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'PS-Auth key=' + API_KEY + '; ' + 'runas=' + USERNAME + '; ' + 'pwd=[' + PASSWORD + '];'
+}
+
+SESSION = requests.session()
+# Remove proxy if not set to true in params
+if not demisto.params().get('proxy'):
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: use `handle_proxy()` instead

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+    if (demisto.params()['url'] and demisto.params()['url'].endswith('/')) else demisto.params()['url']
+# Should we use SSL
+USE_SSL = not demisto.params().get('insecure', False)
+# Service base URL
+BASE_URL = SERVER + '/BeyondTrust/api/public/v3'
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'PS-Auth key=' + API_KEY + '; ' + 'runas=' + USERNAME + '; ' + 'pwd=[' + PASSWORD + '];'
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: use `f'...'` instead of `+`

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 5]
```diff
@@ -0,0 +1,316 @@
...
+    del os.environ['HTTP_PROXY']
+    del os.environ['HTTPS_PROXY']
+    del os.environ['http_proxy']
+    del os.environ['https_proxy']
+
+
+''' HELPER FUNCTIONS '''
+
+
+def http_request(method, suffix_url, params=None, data=None):
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: need to catch here connection errors

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 6]
```diff
@@ -0,0 +1,316 @@
...
+        method,
+        url,
+        verify=USE_SSL,
+        params=params,
+        data=data,
+        headers=HEADERS
+    )
+    # Handle error responses gracefully
+    if res.status_code not in {200, 201, 204}:
+        return_error('Error in API call to BeyondSafe Integration [%d] - %s' % (res.status_code, res.content))
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: use `f'...'`

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 7]
```diff
@@ -0,0 +1,316 @@
...
+    except ValueError:
+        return None
+
+
+def signin():
+
+    suffix_url = '/Auth/SignAppin'
+    header = {'Authorization': 'PS-Auth key=' + API_KEY + '; ' + 'runas=' + USERNAME + '; ' + 'pwd=[' + PASSWORD + '];'}
+    SESSION.headers.update(header)
+    response = SESSION.post(BASE_URL + suffix_url,verify=USE_SSL)
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why not use `http_request` ?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 8]
```diff
@@ -0,0 +1,316 @@
...
+
+
+def signout():
+
+    suffix_url = '/auth/signout'
+    response = SESSION.post(BASE_URL + suffix_url)
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why not use `http_request` ?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 9]
```diff
@@ -0,0 +1,316 @@
...
+
+
+''' COMMANDS + REQUESTS FUNCTIONS '''
+
+
+def test_module():
+    """
+    Singing in the execution code
+    """
+    pass
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: ??

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 10]
```diff
@@ -0,0 +1,316 @@
...
+def get_managed_accounts():
+    """
+    Returns a list of Managed Accounts that can be requested by the current user.
+    """
+    data = []
+    managed_accounts = get_managed_accounts_request()
+    for account in managed_accounts:
+        data.append({
+            'PlatformID': account.get('PlatformID'),
+            'SystemId': account.get('SystemId'),
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
            'SystemID': account.get('SystemId'),
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 11]
```diff
@@ -0,0 +1,316 @@
...
+            'SystemName': account.get('SystemName'),
+            'DomainName': account.get('DomainName'),
+            'AccountId': account.get('AccountId'),
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
            'AccountID': account.get('AccountId'),
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 12]
```diff
@@ -0,0 +1,316 @@
...
+    return response
+
+
+def create_release():
+    """
+    Creates a new release request.
+    Retrieves the credentials for an approved and active (not expired) credentials release request.
+    """
+    access_type = demisto.args().get('access_type')
+    system_id = demisto.args().get('system_id').encode('utf-8')
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why is the encoding necessary here ?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 13]
```diff
@@ -0,0 +1,316 @@
...
+
+def http_request(method, suffix_url, params=None, data=None):
+
+    url = BASE_URL + suffix_url
+
+    res = SESSION.request(
+        method,
+        url,
+        verify=USE_SSL,
+        params=params,
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: you are not using `params`, remove if not needed

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 14]
```diff
@@ -0,0 +1,316 @@
...
+        data=data,
+        headers=HEADERS
+    )
+    # Handle error responses gracefully
+    if res.status_code not in {200, 201, 204}:
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: are all these status codes needed ?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 15]
```diff
@@ -0,0 +1,316 @@
...
+
+    credentials = get_credentials_request(request_id)
+
+    response = {
+        'RequestID': request_id,
+        'Credentials': credentials
+    }
+
+    ec = {'BeyondTrust.Request(val.AccountId === obj.AccountId)':createContext(response)}
+    return_outputs(tableToMarkdown('The request has been Successful', response), ec, response)
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: improve title

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 16]
```diff
@@ -0,0 +1,316 @@
...
+    request_id = str(request)
+
+    credentials = get_credentials_request(request_id)
+
+    response = {
+        'RequestID': request_id,
+        'Credentials': credentials
+    }
+
+    ec = {'BeyondTrust.Request(val.AccountId === obj.AccountId)':createContext(response)}
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: ad white space after
```suggestion
    ec = {'BeyondTrust.Request(val.AccountId === obj.AccountId)': createContext(response)}
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 17]
```diff
@@ -0,0 +1,316 @@
...
+
+def get_credentials_request(request_id):
+
+    suffix_url = '/credentials/' + request_id
+    response = http_request('GET', suffix_url)
+
+    return response
+
+
+# def get_credentials():
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: ??

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 18]
```diff
@@ -0,0 +1,316 @@
...
+#     response = get_credentials_request(request_id)
+
+#     ec = {'BeyondTrust.Request(val.request_id === obj.request_id)': response}
+
+#     return_outputs(tableToMarkdown('BeyondTrust Release Request', response, removeNull=True), ec, response)
+
+
+def check_in_credentials_request(request_id, data):
+
+    suffix_url = '/Requests/' + request_id + '/Checkin'
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: use `f'...'`

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 19]
```diff
@@ -0,0 +1,316 @@
...
+    data = {'Reason': reason if reason else ''}
+
+    check_in = check_in_credentials_request(request_id, data)
+
+    demisto.results('The release was successfully checked-in/released')
+
+
+def change_credentials_request(account_id, data):
+
+    suffix_url = '/ManagedAccounts/' + account_id + '/Credentials'
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: use `f'...'`

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 20]
```diff
@@ -0,0 +1,316 @@
...
+
+    if password:
+        data['Password'] = password
+
+    if private_key:
+        if public_key and update_system == True:
+            data['PrivateKey'] = private_key
+            data['PublicKey'] = public_key
+        else:
+            demisto.results('Please provide public key')
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: dont we want to exit the function after that?
also i think we should return an error here
dont use `Please` please :)

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 21]
```diff
@@ -0,0 +1,316 @@
...
+# Service base URL
+BASE_URL = SERVER + '/BeyondTrust/api/public/v3'
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+    'Authorization': 'PS-Auth key=' + API_KEY + '; ' + 'runas=' + USERNAME + '; ' + 'pwd=[' + PASSWORD + '];'
+}
+
+SESSION = requests.session()
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why are we using a session?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 22]
```diff
@@ -0,0 +1,316 @@
...
+  required: true
+  type: 4
+- display: Fetch credentials
+  name: isFetchCredentials
+  required: false
+  type: 8
+description: Unified password and session management for seamless accountability and
+  control over privileged accounts.
+display: BeyondTrust_Password_Safe
+name: BeyondTrust_Password_Safe
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: remove under
```suggestion
name: BeyondTrust Password Safe
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 23]
```diff
@@ -0,0 +1,316 @@
...
+  name: key
+  required: true
+  type: 4
+- display: Fetch credentials
+  name: isFetchCredentials
+  required: false
+  type: 8
+description: Unified password and session management for seamless accountability and
+  control over privileged accounts.
+display: BeyondTrust_Password_Safe
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]:
```suggestion
display: BeyondTrust Password Safe
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 24]
```diff
@@ -0,0 +1,316 @@
+category: Endpoint
+commonfields:
+  id: BeyondTrust_Password_Safe
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]:
```suggestion
  id: BeyondTrust Password Safe
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 25]
```diff
@@ -0,0 +1,316 @@
...
+- display: Server URL
+  name: url
+  required: true
+  type: 0
+- defaultvalue: 'true'
+  display: Trust any certificate (not secure)
+  name: insecure
+  required: false
+  type: 8
+- display: Use system proxy
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]:
```suggestion
- display: Use system proxy settings
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 26]
```diff
@@ -0,0 +1,316 @@
+category: Endpoint
+commonfields:
+  id: BeyondTrust_Password_Safe
+  version: -1
+configuration:
+- display: Server URL
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: add example
no default value?

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 27]
```diff
@@ -0,0 +1,316 @@
...
+  name: url
+  required: true
+  type: 0
+- defaultvalue: 'true'
+  display: Trust any certificate (not secure)
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: insecure and proxy should be below credentials and api key

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 28]
```diff
@@ -0,0 +1,316 @@
+import demistomock as demisto
+from CommonServerPython import *
+from CommonServerUserPython import *
+''' IMPORTS '''
+
+import json
+import requests
+from requests import Request, Session
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why are you importing `Request` for?
don't import `requests` and `from requests`

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py (hunk 14 above)
Comment (Shellyber) [reply Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: yes, I have a few commands that return 201 or 204 as a Success status code

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py (hunk 21 above)
Comment (Shellyber) [reply Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: We need to use Session According to the API guide

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 29]
```diff
@@ -0,0 +1,336 @@
...
+      isArray: false
+      name: request_id
+      required: true
+      secret: false
+    deprecated: false
+    description: Retrieves the credentials for an approved and active (not expired)
+      credentials release request.
+    execution: false
+    name: beyondtrust-get-credentials
+  dockerimage: demisto/python3
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: use latest tag
```suggestion
  dockerimage: demisto/python3:3.7.3.286
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 30]
```diff
@@ -0,0 +1,366 @@
...
+            'NextChangeDate': account.get('NextChangeDate'),
+            'SystemID': account.get('SystemId'),
+            'SystemName': account.get('SystemName'),
+            'DomainName': account.get('DomainName'),
+            'AccountID': account.get('AccountId'),
+            'AccountName': account.get('AccountName')
+
+        })
+
+    entry_context = {'BeyondTrust.Account(val.AccountId === obj.AccountId)': managed_accounts}
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
    entry_context = {'BeyondTrust.Account(val.AccountID === obj.AccountID)': managed_accounts}
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 31]
```diff
@@ -0,0 +1,366 @@
...
+    request_id = str(request)
+
+    credentials = get_credentials_request(request_id)
+
+    response = {
+        'RequestID': request_id,
+        'Password': credentials
+    }
+
+    entry_context = {'BeyondTrust.Request(val.AccountId === obj.AccountId)': createContext(response)}
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
    entry_context = {'BeyondTrust.Request(val.AccountID === obj.AccountID)': createContext(response)}
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 32]
```diff
@@ -0,0 +1,336 @@
+category: Authentication
+commonfields:
+  id: BeyondTrust Password Safe
+  version: -1
+configuration:
+- display: Server URL (e.g https://ec2-54-194-234-104.eu-west-1.compute.amazonaws.com)
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: change example url

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 33]
```diff
@@ -0,0 +1,394 @@
...
+# Service base URL
+BASE_URL = SERVER + '/BeyondTrust/api/public/v3'
+# Headers to be sent in requests
+HEADERS = {
+    'Content-Type': 'application/json',
+    'Accept': 'application/json',
+}
+
+SESSION = requests.session()
+RAISE_EXCEPTION_ON_ERROR: bool = False
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]: why is this needed?

File: TestPlaybooks/playbook-BeyondTrust-test.yml [hunk 34]
```diff
@@ -0,0 +1,217 @@
...
+      access_policy_schedule_id: {}
+      access_type: {}
+      account_id:
+        complex:
+          root: BeyondTrust
+          filters:
+          - - operator: isEqualString
+              left:
+                value:
+                  simple: BeyondTrust.Account.AccountId
```
Comment (Itay4) [inline TestPlaybooks/playbook-BeyondTrust-test.yml]:
```suggestion
                  simple: BeyondTrust.Account.AccountID
```

File: TestPlaybooks/playbook-BeyondTrust-test.yml [hunk 35]
```diff
@@ -0,0 +1,217 @@
...
+                iscontext: true
+              right:
+                value:
+                  simple: "6"
+          accessor: Account.AccountId
```
Comment (Itay4) [inline TestPlaybooks/playbook-BeyondTrust-test.yml]:
```suggestion
          accessor: Account.AccountID
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 36]
```diff
@@ -0,0 +1,330 @@
...
+      description: ID of the managed system platform.
+      type: Number
+    - contextPath: BeyondTrust.Account.SystemId
+      description: ID of the managed system.
+      type: Number
+    - contextPath: BeyondTrust.Account.SystemName
+      description: Name of the managed system.
+      type: String
+    - contextPath: BeyondTrust.Account.DomainName
+      description: 'ID of the managed account. '
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: remove white space

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 37]
```diff
@@ -0,0 +1,330 @@
...
+      type: Number
+    - contextPath: BeyondTrust.Account.AccountName
+      description: Name of the managed account.
+      type: String
+    - contextPath: BeyondTrust.Account.InstanceName
+      description: 'Database instance name of a database-type managed system. '
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: remove white space

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 38]
```diff
@@ -0,0 +1,330 @@
...
+    - contextPath: BeyondTrust.Account.NexeChangeDate
+      description: The date and time of the next scheduled password change.
+      type: Date
+    - contextPath: BeyondTrust.Account.IsChanging
+      description: True if the account credentials are in the process of changing,
+        otherwise false.
+      type: Boolean
+    - contextPath: BeyondTrust.Account.IsISAAccess
+      description: 'True if the account is for Information Systems Administrator (ISA)
+        access, otherwise false. '
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: remove white space

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml [hunk 39]
```diff
@@ -0,0 +1,330 @@
...
+      type: Boolean
+    - contextPath: BeyondTrust.Account.AccountId
+      description: 'ID of the managed account. '
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.yml]: remove white space

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 40]
```diff
@@ -0,0 +1,394 @@
...
+    Returns a list of Managed Accounts that can be requested by the current user.
+    """
+    data = []
+    headers = ['AccountName', 'AccountID', 'AssetName', 'SystemID', 'DomainName', 'LastChangeDate', 'NextChangeDate']
+    managed_accounts = get_managed_accounts_request()
+    for account in managed_accounts:
+        data.append({
+            'LastChangeDate': account.get('LastChangeDate'),
+            'NextChangeDate': account.get('NextChangeDate'),
+            'SystemID': account.get('SystemId'),
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
            'AssetID': account.get('SystemId'),
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 41]
```diff
@@ -0,0 +1,394 @@
...
+    data = []
+    managed_systems = get_managed_systems_request()
+    for managed_system in managed_systems:
+        data.append({
+            'ManagedSystemID': managed_system.get('ManagedSystemID'),
+            'ChangeFrequencyDays': managed_system.get('ChangeFrequencyDays'),
+            'AssetID': managed_system.get('AssetID'),
+            'DatabaseID': managed_system.get('DatabaseID'),
+            'DirectoryID': managed_system.get('DirectoryID'),
+            'SystemName': managed_system.get('SystemName'),
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
            'AssetName': managed_system.get('SystemName'),
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 42]
```diff
@@ -0,0 +1,394 @@
...
+
+def get_managed_systems():
+    """
+    Returns a list of Managed Systems.
+    """
+    data = []
+    managed_systems = get_managed_systems_request()
+    for managed_system in managed_systems:
+        data.append({
+            'ManagedSystemID': managed_system.get('ManagedSystemID'),
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
            'ManagedAssetID': managed_system.get('ManagedSystemID'),
```

File: Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py [hunk 43]
```diff
@@ -0,0 +1,398 @@
...
+            'ChangeFrequencyDays': managed_system.get('ChangeFrequencyDays'),
+            'AssetID': managed_system.get('AssetID'),
+            'DatabaseID': managed_system.get('DatabaseID'),
+            'DirectoryID': managed_system.get('DirectoryID'),
+            'AssetName': managed_system.get('SystemName'),
+            'PlatformID': managed_system.get('PlatformID'),
+            'Port': managed_system.get('Port')
+        })
+
+    entry_context = {'BeyondTrust.System(val.ManagedSystemID === obj.ManagedSystemID)': managed_systems}
```
Comment (Itay4) [inline Integrations/BeyondTrust_Password_Safe/BeyondTrust_Password_Safe.py]:
```suggestion
    entry_context = {'BeyondTrust.System(val. ManagedAssetID === obj. ManagedAssetID)': managed_systems}
```
